ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGetLogs.java

```diff
@@ -71,4 +72,26 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     return new JsonRpcSuccessResponse(
         requestContext.getRequest().getId(), new LogsResult(matchingLogs));
   }
+
+  private long interpretBlockParam(final BlockParameter block) {
+    if (block.isFinalized()) {
+      return blockchain
+          .finalizedBlockHeader()
+          .orElseThrow(() -> new IllegalArgumentException("Finalized block not found."))
```

Comment:
IllegalArg? will this return a JsonErrorResponse to the user

Reply:
Not quite. Will address...